lib/service/service.go

```diff
@@ -1674,9 +1677,10 @@ func (process *TeleportProcess) initAuthService() error {
 
 	// use multiplexer to leverage support for proxy protocol.
 	mux, err := multiplexer.New(multiplexer.Config{
-		EnableProxyProtocol: cfg.Auth.EnableProxyProtocol,
-		Listener:            listener,
-		ID:                  teleport.Component(process.id),
+		EnableExternalProxyProtocol: cfg.Auth.EnableProxyProtocol,
+		Listener:                    listener,
+		ID:                          teleport.Component(process.id),
+		CertAuthoritiesGetter:       authCache,
```

Comment:
Are we ok with disabling the entire feature if caching is disabled on the auth?

Reply:
Good point. So, I'll pass `authServer.Services` then, if cache is not initialized. 
What's the production use case for disabling cache btw?

Reply:
I'd just pass `authServer` and let that figure it out.